code/game/gamemodes/steal_items.dm

```diff
@@ -35,30 +35,25 @@
 	return L
 
 /datum/theft_objective/proc/check_completion(datum/mind/owner)
-	//testing("Checking completion for [src.name] ([typepath])")
 	if (owner && owner.current)
-		//testing("Mind datum([owner]) and mob([owner.current]) is good.")
-		var/list/all_items = new/list()
+		var/all_items = new/list()
 
 		if (isliving(owner.current))
 			all_items += get_contents_in_object(owner.current)
-		//testing("Got [all_items.len] objects in [owner.current]")
 
-		if(areas && areas.len)
-			for (var/area_type in areas)
-				all_items += get_contents_in_object(locate(area_type), /obj)
-		//testing("Starting to check the list for [typepath]")
+		for (var/area_type in areas)
```

Comment:
There's no need for if(areas && areas.len) unless it's the area of var you're using in for loop